File: reports/20260826-173422/results/service-bus/data-plane/js-ts/crud/service-bus-dp-js-ts-crud/js-ts-azure-tools/baseline/generated-code/src/service-bus-demo.ts
import {
  ProcessErrorArgs,
  ServiceBusClient,
  ServiceBusMessage,
} from "@azure/service-bus";

const requiredEnvironmentVariable = (name: string): string => {
  const value = process.env[name];
  if (!value) {
    throw new Error(`Set the ${name} environment variable before running this example.`);
  }
  return value;
};

const delay = (milliseconds: number): Promise<void> =>
  new Promise((resolve) => setTimeout(resolve, milliseconds));

async function main(): Promise<void> {
  const connectionString = requiredEnvironmentVariable(
    "AZURE_SERVICE_BUS_CONNECTION_STRING",
  );
  const queueName = requiredEnvironmentVariable("AZURE_SERVICE_BUS_QUEUE_NAME");
  const topicName = requiredEnvironmentVariable("AZURE_SERVICE_BUS_TOPIC_NAME");
  const subscriptionName = requiredEnvironmentVariable(
    "AZURE_SERVICE_BUS_SUBSCRIPTION_NAME",
  );

  // The queue, topic, and subscription must already exist.
  const client = new ServiceBusClient(connectionString);

  try {
    const queueSender = client.createSender(queueName);
    try {
      await queueSender.sendMessages({
        body: "Single queue message",
        contentType: "text/plain",
      });

      const batch = await queueSender.createMessageBatch();
      for (let sequence = 1; sequence <= 5; sequence += 1) {
        const message: ServiceBusMessage = {
          body: { sequence, text: `Batch message ${sequence}` },
          contentType: "application/json",
          messageId: `batch-${sequence}`,
        };

        if (!batch.tryAddMessage(message)) {
          throw new Error(`Batch message ${sequence} is too large for the batch.`);
        }
      }
      await queueSender.sendMessages(batch);
    } finally {
      await queueSender.close();
    }

    const queueReceiver = client.createReceiver(queueName);
    try {
      const messages = await queueReceiver.receiveMessages(6, {
        maxWaitTimeInMs: 10_000,
      });

      for (const message of messages) {
        console.log("Received from queue:", message.body);
        await queueReceiver.completeMessage(message);
      }
    } finally {
      await queueReceiver.close();
    }

    const subscriptionReceiver = client.createReceiver(queueName);
    const subscription = subscriptionReceiver.subscribe({
      processMessage: async (message) => {
        console.log("Subscription handler received:", message.body);
        await subscriptionReceiver.completeMessage(message);
      },
      processError: async (args: ProcessErrorArgs) => {
        console.error(
          `Subscription error from ${args.errorSource}:`,
          args.error,
        );
      },
    });

    try {
      // Keep the handler active briefly for this finite demo.
      await delay(10_000);
    } finally {
      await subscription.close();
      await subscriptionReceiver.close();
    }

    const topicSender = client.createSender(topicName);
    try {
      await topicSender.sendMessages({
        body: "Message sent to a topic",
        subject: "topic-demo",
      });
    } finally {
      await topicSender.close();
    }

    const topicReceiver = client.createReceiver(topicName, subscriptionName);
    try {
      const [topicMessage] = await topicReceiver.receiveMessages(1, {
        maxWaitTimeInMs: 10_000,
      });

      if (topicMessage) {
        console.log("Received from topic subscription:", topicMessage.body);
        await topicReceiver.completeMessage(topicMessage);
      } else {
        console.log("No topic message arrived before the receive timeout.");
      }
    } finally {
      await topicReceiver.close();
    }
  } finally {
    await client.close();
  }
}

main().catch((error: unknown) => {
  console.error("Service Bus demo failed:", error);
  process.exitCode = 1;
});

File: rerun-reports/lane-a/027/20260828-023618/results/service-bus/data-plane/java/crud/service-bus-dp-java-crud/java-azure-skills/azure-skill-mcp-microsoft-skill/generated-code/src/main/java/com/example/ServiceBusDemo.java
package com.example;

import com.azure.messaging.servicebus.ServiceBusClientBuilder;
import com.azure.messaging.servicebus.ServiceBusErrorContext;
import com.azure.messaging.servicebus.ServiceBusMessage;
import com.azure.messaging.servicebus.ServiceBusMessageBatch;
import com.azure.messaging.servicebus.ServiceBusProcessorClient;
import com.azure.messaging.servicebus.ServiceBusReceivedMessage;
import com.azure.messaging.servicebus.ServiceBusReceiverClient;
import com.azure.messaging.servicebus.ServiceBusSenderClient;
import com.azure.messaging.servicebus.models.ServiceBusReceiveMode;

import java.time.Duration;
import java.util.concurrent.CountDownLatch;
import java.util.concurrent.TimeUnit;

public final class ServiceBusDemo {
    private static final String CONNECTION_STRING_ENV = "AZURE_SERVICE_BUS_CONNECTION_STRING";
    private static final String QUEUE_NAME_ENV = "AZURE_SERVICE_BUS_QUEUE_NAME";
    private static final String TOPIC_NAME_ENV = "AZURE_SERVICE_BUS_TOPIC_NAME";
    private static final String SUBSCRIPTION_NAME_ENV = "AZURE_SERVICE_BUS_SUBSCRIPTION_NAME";

    private ServiceBusDemo() {
    }

    public static void main(String[] args) throws InterruptedException {
        String connectionString = requiredEnvironmentVariable(CONNECTION_STRING_ENV);
        String queueName = requiredEnvironmentVariable(QUEUE_NAME_ENV);
        String topicName = requiredEnvironmentVariable(TOPIC_NAME_ENV);
        String subscriptionName = requiredEnvironmentVariable(SUBSCRIPTION_NAME_ENV);

        sendToQueue(connectionString, queueName);
        receiveFromQueue(connectionString, queueName);
        processQueueContinuously(connectionString, queueName);
        demonstrateTopicSubscription(connectionString, topicName, subscriptionName);
    }

    private static void sendToQueue(String connectionString, String queueName) {
        ServiceBusSenderClient sender = new ServiceBusClientBuilder()
            .connectionString(connectionString)
            .sender()
            .queueName(queueName)
            .buildClient();

        try {
            sender.sendMessage(new ServiceBusMessage("Single queue message"));
            System.out.println("Sent one queue message.");

            ServiceBusMessageBatch batch = sender.createMessageBatch();
            for (int i = 1; i <= 5; i++) {
                ServiceBusMessage message = new ServiceBusMessage("Batch queue message " + i)
                    .setMessageId("batch-" + i);
                if (!batch.tryAddMessage(message)) {
                    throw new IllegalArgumentException(
                        "Batch message " + i + " is too large for the Service Bus batch.");
                }
            }

            sender.sendMessages(batch);
            System.out.printf("Sent a batch of %d queue messages.%n", batch.getCount());
        } finally {
            sender.close();
        }
    }

    private static void receiveFromQueue(String connectionString, String queueName) {
        ServiceBusReceiverClient receiver = new ServiceBusClientBuilder()
            .connectionString(connectionString)
            .receiver()
            .queueName(queueName)
            .receiveMode(ServiceBusReceiveMode.PEEK_LOCK)
            .buildClient();

        try {
            for (ServiceBusReceivedMessage message
                : receiver.receiveMessages(6, Duration.ofSeconds(30))) {
                System.out.printf("Queue receiver processed: %s%n", message.getBody());
                receiver.complete(message);
            }
        } finally {
            receiver.close();
        }
    }

    private static void processQueueContinuously(String connectionString, String queueName)
        throws InterruptedException {
        CountDownLatch messageProcessed = new CountDownLatch(1);

        ServiceBusProcessorClient processor = new ServiceBusClientBuilder()
            .connectionString(connectionString)
            .processor()
            .queueName(queueName)
            .receiveMode(ServiceBusReceiveMode.PEEK_LOCK)
            .disableAutoComplete()
            .processMessage(context -> {
                ServiceBusReceivedMessage message = context.getMessage();
                System.out.printf("Processor handled: %s%n", message.getBody());
                context.complete();
                messageProcessed.countDown();
            })
            .processError(ServiceBusDemo::processError)
            .buildProcessorClient();

        ServiceBusSenderClient sender = new ServiceBusClientBuilder()
            .connectionString(connectionString)
            .sender()
            .queueName(queueName)
            .buildClient();

        try {
            processor.start();
            sender.sendMessage(new ServiceBusMessage("Message for the continuous processor"));

            if (!messageProcessed.await(30, TimeUnit.SECONDS)) {
                System.out.println("The processor did not receive a message within 30 seconds.");
            }
        } finally {
            sender.close();
            processor.stop();
            processor.close();
        }
    }

    private static void demonstrateTopicSubscription(
        String connectionString,
        String topicName,
        String subscriptionName) {

        ServiceBusSenderClient topicSender = new ServiceBusClientBuilder()
            .connectionString(connectionString)
            .sender()
            .topicName(topicName)
            .buildClient();

        ServiceBusReceiverClient subscriptionReceiver = new ServiceBusClientBuilder()
            .connectionString(connectionString)
            .receiver()
            .topicName(topicName)
            .subscriptionName(subscriptionName)
            .receiveMode(ServiceBusReceiveMode.PEEK_LOCK)
            .buildClient();

        try {
            topicSender.sendMessage(new ServiceBusMessage("Topic message"));

            for (ServiceBusReceivedMessage message
                : subscriptionReceiver.receiveMessages(1, Duration.ofSeconds(30))) {
                System.out.printf("Subscription received: %s%n", message.getBody());
                subscriptionReceiver.complete(message);
            }
        } finally {
            subscriptionReceiver.close();
            topicSender.close();
        }
    }

    private static void processError(ServiceBusErrorContext context) {
        System.err.printf(
            "Processor error from %s in namespace %s: %s%n",
            context.getErrorSource(),
            context.getFullyQualifiedNamespace(),
            context.getException().getMessage());
    }

    private static String requiredEnvironmentVariable(String name) {
        String value = System.getenv(name);
        if (value == null || value.isBlank()) {
            throw new IllegalStateException("Set the " + name + " environment variable.");
        }
        return value;
    }
}
